src/data/simple_dmatrix.cc

```diff
@@ -47,13 +48,29 @@ class SimpleBatchIteratorImpl : public BatchIteratorImpl {
   }
 
  private:
-  SparsePage* page_{nullptr};
+  T* page_{nullptr};
 };
 
+BatchSet SimpleDMatrix::GetBatches(PageType page_type) {
+  switch(page_type) {
+    case kCSR:
+      return GetRowBatches();
+    case kCSC:
+      return GetColumnBatches();
+    case kSortedCSC:
+      return GetSortedColumnBatches();
+    default:
+      LOG(FATAL) << "Unknown page type";
+      return BatchSet(nullptr);
```

Comment:
is this a no-op? or is this required because compiler complains about not returning something in some code paths?